python/ray/dag/tests/experimental/test_accelerated_dag.py

```diff
@@ -2579,6 +2680,122 @@ def echo(self, x):
 
         compiled_dag.teardown()
 
+    def test_visualize_complicate(self, ray_start_regular):
+        """
+        Expect output:
+            Nodes Information:
+            0 [label="Task 0
+            InputNode"]
+            1 [label="Task 1
+            Actor: 037b25...
+            Method: return_three"]
+            2 [label="Task 2
+            ClassMethodOutputNode[0]"]
+            3 [label="Task 3
+            ClassMethodOutputNode[1]"]
+            4 [label="Task 4
+            ClassMethodOutputNode[2]"]
+            5 [label="Task 5
+            Actor: f1130d...
+            Method: echo"]
+            6 [label="Task 6
+            Actor: f1130d...
+            Method: echo"]
+            7 [label="Task 7
+            Actor: f1130d...
+            Method: return_two"]
+            8 [label="Task 8
+            MultiOutputNode"]
+            9 [label="Task 9
+            ClassMethodOutputNode[0]"]
+            10 [label="Task 10
+            ClassMethodOutputNode[1]"]
+
+            Edges Information:
+            0 -> 1 [label=SharedMemoryType]
+            1 -> 2 [label=SharedMemoryType]
+            1 -> 3 [label=SharedMemoryType]
+            1 -> 4 [label=SharedMemoryType]
+            2 -> 5 [label=SharedMemoryType]
+            3 -> 6 [label=SharedMemoryType]
+            4 -> 7 [label=SharedMemoryType]
+            5 -> 8 [label=SharedMemoryType]
+            6 -> 8 [label=SharedMemoryType]
+            9 -> 8 [label=ChannelOutputType]
+            10 -> 8 [label=ChannelOutputType]
+            7 -> 9 [label=SharedMemoryType]
+            7 -> 10 [label=SharedMemoryType]
+
+            Experimental Graph Built:
+            0:InputNode
+            |
+            1:Actor:return_three
+            |-------------------|-------------------|
+            2:Output[0]         3:Output[1]         4:Output[2]
+            |                   |                   |
+            5:Actor:echo        6:Actor:echo        7:Actor:return_two
```

Comment:
Can you add a test case where there are crossing of the edges? I feel it may not look well since the edges are not directed right now.

Reply:
Is this addressed?

Reply:
I add directions now